### PR TITLE
Refactor database configuration and migration

### LINKED ISSUE #112: Skift fra SQLite til Postgres
Husk at tage api ned så vi ikke har brugere der ikke bliver flyttet med

### PR DESCRIPTION
**Pull request by AeselCSS**
## Description

This pull request refactors the database configuration and migration logic in the codebase. It introduces support for connecting to a Postgres database and removes the SQLite database files from the repository. The `InitDatabase` function now connects to a Postgres database using the configuration values from `config.AppConfig.Database`. The `InitTestDatabase` function establishes a connection to an SQLite in-memory database for testing purposes. The `autoMigrate` function is used to automatically migrate the database schema to the latest version, including the `User`, `Page`, and `JWT` models.

### Issues # (issue-id)
Fixes #112, #126, #127, #132, #150

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code quality improvements (linting, refactoring, etc.)
- [ ] Other (please specify):

## Checklist

- [x] My code follows the style guidelines of this project (ESLint, GoLint, etc.).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have run relevant linters (ESLint, GoLint, Qodana, etc.) and ensured that no new issues were introduced.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -96,69 +95,74 @@ jobs:
...
+    - name: Deploy to Azure VM
+      run: |
+        echo "Deploying with IMAGE_TAG: ${{ env.IMAGE_TAG }}"
+
+        # SSH into the Azure VM and deploy
+        ssh -o StrictHostKeyChecking=no ${{ secrets.AZURE_VM_USER }}@${{ secrets.AZURE_VM_IP }} << EOF
+          cd /home/azureuser/whoknows || exit
+
+          # Optionally pull the compose.yml if needed
+          git fetch origin main
```
Comment (emilvn) [inline .github/workflows/deploy.yml:113]: Nice!

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -96,69 +95,74 @@ jobs:
...
+
+          # Pull the latest images from GHCR
+          echo "Pulling backend image ghcr.io/${{ github.repository }}/backend:${{ env.IMAGE_TAG }}"
+          docker pull ghcr.io/${{ github.repository }}/backend:${{ env.IMAGE_TAG }}
+
+          echo "Pulling frontend image ghcr.io/${{ github.repository }}/frontend:${{ env.IMAGE_TAG }}"
+          docker pull ghcr.io/${{ github.repository }}/frontend:${{ env.IMAGE_TAG }}
+
+          # Build and start Docker services with the 'production' profile
+          echo "Building and starting Docker services with the production profile..."
```
Comment (emilvn) [inline .github/workflows/deploy.yml:141]: Teknisk set builder vi vel ikke her? vi starter vel bare containerne baseret på de images vi har fetchet

File: backend/internal/config/config.go [hunk 3]
```diff
@@ -8,100 +8,101 @@ import (
...
+
+		// Weather API Configuration
+		"API_WEATHER_API_KEY": func() error { AppConfig.WeatherAPI.OpenWeatherAPIKey, err = getEnv("API_WEATHER_API_KEY"); return err },
 	}
 
-	offset, err := strconv.Atoi(getEnv("API_PAGINATION_OFFSET", "0"))
-
-	if err != nil {
-		return fmt.Errorf("invalid PAGINATION_OFFSET value")
+	for key, fn := range loadConfig {
```
Comment (emilvn) [inline backend/internal/config/config.go:71]: Fancy!

File: backend/internal/database/db.go [hunk 4]
```diff
@@ -2,53 +2,79 @@ package database
...
+	dsn := fmt.Sprintf(
+		"host=%s user=%s password=%s dbname=%s port=%d sslmode=%s",
+		config.AppConfig.Database.Host,
+		config.AppConfig.Database.User,
+		config.AppConfig.Database.Password,
+		config.AppConfig.Database.Name,
+		config.AppConfig.Database.Port,
+		config.AppConfig.Database.SSLMode,
+	)
+	fmt.Printf("Attempting to connect to Postgres database with DSN: %s\n", dsn)
```
Comment (emilvn) [inline backend/internal/database/db.go:29]: Vi bør ikke udskrive password til log her

File: backend/test/helpers/test_helpers.go
Comment (emilvn) [inline backend/test/helpers/test_helpers.go:1]: Ryd lidt op i kommentarer i denne fil

File: backend/test/test.env
Comment (emilvn) [inline backend/test/test.env:1]: er det med vilje vi pusher den her med? I så fald, så se bort fra denne kommentar.

File: backend/test/unit/config_test.go
Comment (emilvn) [inline backend/test/unit/config_test.go:1]: Ryd lidt op i kommentarer i denne fil

File: backend/test/unit/jwt_test.go
Comment (emilvn) [inline backend/test/unit/jwt_test.go:1]: Ryd lidt op i kommentarer i denne fil

File: backend/test/unit/user_service_test.go
Comment (emilvn) [inline backend/test/unit/user_service_test.go:1]: Ryd lidt op i kommentarer i denne fil

File: frontend/tests/login.spec.ts
Comment (emilvn) [inline frontend/tests/login.spec.ts:1]: Ryd lidt op i kommentarer i denne fil

**Review by emilvn — CHANGES_REQUESTED** [review]
Overall ser det godt ud, har lidt spørgsmål og forespørgsler som jeg har tilføjet som kommentarer som du gerne lige må kigge på inden vi merger.
image

File: backend/internal/database/db.go (hunk 4 above)
Comment (AeselCSS) [reply backend/internal/database/db.go:29]: du har ret! - det var for debugging og skal naturligvis fjernes

File: backend/test/test.env
Comment (AeselCSS) [reply backend/test/test.env:1]: den bruges af config_test.go

**Review by emilvn — APPROVED** [review]
image

**Comment by emilvn** [comment]
Husk at merge med squash commit

**Pull request merged**